frontend/src/context/Filtercontext.js

```diff
@@ -1,15 +1,16 @@
-import React,{createContext,useReducer} from 'react'
+import React, { createContext, useReducer } from 'react'
 import { FilterReducer } from "../reducer/FilterReducer"
-import { Productdb } from '../api/db/Productdb'
+
 const FilterContext = createContext()
 
-function FilterProvider({children}) {
-  const [state,dispatch] = useReducer(FilterReducer,{ productDb: Productdb })
+function FilterProvider({ children }) {
+
+  const [state, dispatch] = useReducer(FilterReducer, { sortBy:null ,rating:null,category:{keyboards:false,switches:false,others:false} })
   return (
```

Comment:
You can write separate object maybe named as initialState and then pass it down here. Its a personal suggestion you can keep it this way if you like